src/java.base/share/classes/sun/nio/ch/ChannelInputStream.java

```diff
@@ -107,6 +110,94 @@ protected int read(ByteBuffer bb)
         return ChannelInputStream.read(ch, bb, true);
     }
 
+    @Override
+    public byte[] readAllBytes() throws IOException {
+        if (!(ch instanceof SeekableByteChannel))
```

Comment:
I think Pattern matching can be used here to make `117` obsolete. So:
```suggestion
        if (!(ch instanceof SeekableByteChannel sbc))
```

Reply:
Fixed.